ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -690,6 +779,36 @@ static enum pkcs11_rc check_attrs_misc_integrity(struct obj_attrs *head)
 	return PKCS11_CKR_OK;
 }
 
+/*
+ * Check access to object against authentication to token
+ */
+enum pkcs11_rc check_access_attrs_against_token(struct pkcs11_session *session,
+						struct obj_attrs *head)
+{
+	bool private = true;
+
+	switch (get_class(head)) {
+	case PKCS11_CKO_SECRET_KEY:
+	case PKCS11_CKO_PUBLIC_KEY:
+	case PKCS11_CKO_DATA:
+		if (!get_bool(head, PKCS11_CKA_PRIVATE))
+			private = false;
```

Comment:
minor: could be __simplified__ to `private = get_bool(head, PKCS11_CKA_PRIVATE);`